packages/data-provider/src/repository/submittedRepository.ts

```diff
@@ -1,21 +1,29 @@
-import { SQL, and, count, eq } from 'drizzle-orm/sql';
+import { SQL, and, count, eq, inArray, isNull, or, sql } from 'drizzle-orm/sql';
 
 import { NewSubmittedData, SubmittedData, submittedData } from '@overture-stack/lyric-data-model';
 import { BaseDependencies } from '../config/config.js';
 import { ServiceUnavailable } from '../utils/errors.js';
-import { BooleanTrueObject, PaginationOptions, SubmittedDataRepository } from '../utils/types.js';
+import { BooleanTrueObject, PaginationOptions, SubmittedDataResponse } from '../utils/types.js';
 
 const repository = (dependencies: BaseDependencies) => {
 	const LOG_MODULE = 'SUBMITTEDDATA_REPOSITORY';
 	const { db, logger } = dependencies;
 
+	// Column name on the database used to build JSONB query
+	const jsonbColumnName = submittedData.data.name;
+
 	const paginatedColumns: BooleanTrueObject = {
 		entityName: true,
 		data: true,
 		organization: true,
 		isValid: true,
 		systemId: true,
 	};
+
+	// Adding softDelete filter as Drizzle currently doesn't support Soft-Delete option
+	// Important: Make sure all Queries on this repository contains this filter
+	const softDeleteFilter = isNull(submittedData.deletedAt);
```

Comment:
Is there any way to enforce that this is always used? This comment does not protect future devs from making a mistake with this.

Reply:
Currently, Drizzle doesn't support soft-delete natively,  so far I haven't found an alternative way to handle soft-delete queries, other than manually adding this `softDeleteFilter` filter.

Reply:
I mean more like, as a developer every query I write needs to have this filter. How do we ensure that no code is committed that has a query without this filter? Is there some code pattern or abstraction that would help?

What you have here works until some new dev tries to edit this file in 3 years and writes a new query without reading this comment.